layouts/partials/head.html

```diff
@@ -1,46 +1,87 @@
 
-    {{ $darkMode := resources.Get "js/darkmode.js" |  resources.Fingerprint "md5" | resources.Minify }}
-    <script src="{{$darkMode.Permalink}}"></script>
-    {{partial "katex.html" .}}
+  {{ $darkMode := resources.Get "js/darkmode.js" | resources.Fingerprint "md5" |
+  resources.Minify }}
+  <script src="{{$darkMode.Permalink}}"></script>
+  {{partial "katex.html" .}}
 
-    <!--  Preload page vars  -->
-    {{$linkIndex := resources.Get "indices/linkIndex.json" | resources.Fingerprint "md5" | resources.Minify |  }}
-    {{$contentIndex := resources.Get "indices/contentIndex.json" | resources.Fingerprint "md5" | resources.Minify }}
-    <script>
-      const BASE_URL = {{.Site.BaseURL}}
-      const fetchData = Promise.all([
-            fetch("{{ $linkIndex.Permalink }}")
-              .then(data => data.json())
-              .then(data => ({
-                index: data.index,
-                links: data.links,
-              })),
-            fetch("{{ $contentIndex.Permalink }}")
-              .then(data => data.json()),
-          ])
-          .then(([{index, links}, content]) => ({
-            index,
-            links,
-            content,
-          }))
-    </script>
+  <!--  Preload page vars  -->
+  {{$linkIndex := resources.Get "indices/linkIndex.json" | resources.Fingerprint
+  "md5" | resources.Minify | }} {{$contentIndex := resources.Get
+  "indices/contentIndex.json" | resources.Fingerprint "md5" | resources.Minify
+  }}
+  <script>
+    const BASE_URL = {{.Site.BaseURL}}
+    const fetchData = Promise.all([
+          fetch("{{ $linkIndex.Permalink }}")
+            .then(data => data.json())
+            .then(data => ({
+              index: data.index,
+              links: data.links,
+            })),
+          fetch("{{ $contentIndex.Permalink }}")
+            .then(data => data.json()),
+        ])
+        .then(([{index, links}, content]) => ({
+          index,
+          links,
+          content,
+        }))
+  </script>
+  <script type="module">
+    import { router, navigate } from "https://unpkg.com/million/dist/router.mjs";
+    window.navigate = navigate;
+    router(".singlePage");
+    const callback = () => {
+      const draw = () => {
+        const container = document.getElementById("graph-container");
+        // retry if the graph is not ready
+        if (!container) return requestAnimationFrame(draw);
+        // clear the graph in case there is anything within it
+        container.textContent = "";
+
+        drawGraph(
+          {{strings.TrimRight "/" .Page.Permalink}},
+          {{strings.TrimRight "/" .Site.BaseURL}},
+          {{$.Site.Data.graphConfig.paths}},
+          {{$.Site.Data.graphConfig.depth}},
+          {{$.Site.Data.graphConfig.enableDrag}},
+          {{$.Site.Data.graphConfig.enableLegend}},
+          {{$.Site.Data.graphConfig.enableZoom}}
+        );
+      };
+      requestAnimationFrame(draw);
+    };
```

Comment:
why does drawGraph need to be within a requestAnimationFrame? isn't it sufficient to just clear it on page navigation?

Reply:
Calling it normally seems to be a race condition. It will attempt to draw the graph before the "SPA" routing is run, leading to a blank graph screen.

`requestAnimationFrame` allows the drawGraph to pause and wait for the routing to be completed, then draw the graph once the new `graph-container` element exists.